### DISCUSSION TITLE
Override dependency in tests

### DISCUSSION THREAD

**Discussion by nicky-isaacs** [krojew/springtime#12] [Q&A]
Hello!

I am evaluating Springtime as a longtime Guice fan looking for a Rust alternative. So far it looks great! The one thing I couldn't quite work out is how to override parts of the dependency graph in tests. Assume I have the following hypothetical dependencies `FriendService => UserService => SqlClient`, in my test I would like to provide a mock implementation for `SqlClient` and build an instance of `FriendService` and invoke it in my tests. How can I accomplish this in Springtime? I have been fumbling around with the methods in `TypedComponentDefinitionRegistry` (`register_component_typed` sounds relevant but alas I have not been able to crack this)

**Comment by krojew** [comment]
The simplest way would be to mark your mock component as primary, so it will override the real one. I'll admit, overriding might be something to improve.
